google/cloud/documentai_toolbox/wrappers/hocr.py

```diff
@@ -0,0 +1,221 @@
+
+    def to_hocr(self, pidx: int,bidx: int):
+        f = ""
+        f += f"<span class='ocr_carea' id='block_{pidx}_{bidx}' title='{self.bounding_box}'>\n"
+        if self.hocr_paragraphs:
+            for paridx,paragraph in enumerate(self.hocr_paragraphs):
+                f += paragraph.to_hocr(pidx=pidx, bidx=bidx, paridx=paridx)
+        f += "</span>\n"
+        return f
+
+
+
+@dataclasses.dataclass
+class _HocrPage:
+    documentai_page: documentai.Document.Page = dataclasses.field(repr=False)
+    document_text: str = dataclasses.field(repr=False)
+    filename: str = dataclasses.field(repr=False)
+
+    hocr_blocks: List[_HocrBlock] = dataclasses.field(init=False, repr=False)
+    bounding_box: str = dataclasses.field(init=False, repr=False)
+    title: str = dataclasses.field(init=False, repr=False)
+
+    def __post_init__(self):
+        self.hocr_blocks = _load_hocr_blocks(page=(self.documentai_page), document_text=(self.document_text))
+        self.bounding_box = _get_bounding_box(element_with_layout=(self.documentai_page), dimensions=(self.documentai_page.dimension))
+        self.title = 'Document'
+    
+    def to_hocr(self, pidx: int):
+        f = ""
+        f += f"<div class='ocr_page' lang='unknown' title='image \"{self.filename}\";{self.bounding_box}'>\n"
+        if self.hocr_blocks:
+            for bidx, block in enumerate(self.hocr_blocks):
+                f += block.to_hocr(pidx=pidx,bidx=bidx) 
+        f += "</div>\n"
+        return f
+
+
+
+@dataclasses.dataclass
+class _Hocr:
+    documentai_pages: List[documentai.Document.Page] = dataclasses.field(repr=False)
+    documentai_text: List[documentai.Document.Page] = dataclasses.field(repr=False)
+    filename: str = dataclasses.field(repr=False)
+
+    hocr_pages: List[_HocrPage] = dataclasses.field(init=False, repr=False)
+    metadata: str = dataclasses.field(init=False, repr=False)
+
+    def __post_init__(self):
+        self.hocr_pages = _load_pages(pages=(self.documentai_pages), document_text=(self.documentai_text),filename=self.filename)
+
+    def export_hocr(self):
+        f = ""
+        f += "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        f += "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n"
+        f += "<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"unknown\" lang=\"unknown\">\n"
+        f += "<head>\n"
+        f += f"<title>{self.filename}</title>\n"
+        f += "<meta http-equiv=\"Content-Type\" content=\"text/html;charset=utf-8\" />\n"
+        f += "<meta name=\"ocr-system\" content=\"Document AI OCR\" />\n"
+        f += "<meta name=\"ocr-langs\" content=\"unknown\" />\n"
+        f += f"<meta name=\"ocr-number-of-pages\" content=\"{len(self.documentai_pages)}\" />\n"
+        f += "<meta name=\"ocr-capabilities\" content=\"ocr_page ocr_carea ocr_par ocr_line ocrx_word\" />\n"
+        f += "</head>\n"
+        f += "<body>\n"
```

Comment:
Put everything that is constant (does not depend on the actual pages) into a single multiline f-string.

Reply:
will do.